dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/InactiveCdiTest.java

```diff
@@ -0,0 +1,88 @@
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+import componenttest.topology.utils.HttpRequest;
+import io.openliberty.mcp.internal.fat.tool.inactiveCdiApp.InactiveCdiTool;
+
+@RunWith(FATRunner.class)
+public class InactiveCdiTest extends FATServletClient {
+    @Server("mcp-server")
+    public static LibertyServer server;
+
+    private static final String EXPECTED_ERROR_MESSAGE = "SRVE0190E: File not found: /mcp";
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, "inactiveCdiTest.war").addPackage(InactiveCdiTool.class.getPackage());
+        ShrinkHelper.exportDropinAppToServer(server, war, SERVER_ONLY);
+        server.startServer();
+    }
+
+    @AfterClass
+    public static void teardown() throws Exception {
+        server.stopServer(EXPECTED_ERROR_MESSAGE);
+    }
+
+    @Test
+    public void testMcpCallWithoutCDIReturnsNotFoundError() throws Exception {
+        String initializeRequest = """
+                        {
+                          "jsonrpc": "2.0",
+                          "id": "1",
+                          "method": "initialize",
+                          "params": {
+                            "protocolVersion": "2025-06-18",
+                            "capabilities": {
+                              "roots": {
+                                "listChanged": true
+                              },
+                              "sampling": {},
+                              "elicitation": {}
+                            },
+                            "clientInfo": {
+                              "name": "FAT Test Client",
+                              "title": "FAT Test Client",
+                              "version": "1.0.0"
+                            }
+                          }
+                        }
+                        """;
+
+        new HttpRequest(server, "/inactiveCdiTest/mcp")
+                                                       .requestProp(ACCEPT, VALUE_ACCEPT_DEFAULT)
+                                                       .requestProp(MCP_PROTOCOL_VERSION, VALUE_MCP_PROTOCOL_VERSION)
+                                                       .jsonBody(initializeRequest)
+                                                       .method("POST")
+                                                       .expectCode(404)
+                                                       .run(String.class);
+        assertNotNull(server.waitForStringInLogUsingMark(EXPECTED_ERROR_MESSAGE, server.getDefaultLogFile())); // expect a file not found error in the logs
```

Comment:
Also test that your warning is logged at startup (i.e. before you make the request to the server)